revm-context: add `impl std::error::Error for TxEnvBuildError`

```
error[E0277]: `?` couldn't convert the error: `revm::revm_context::tx::TxEnvBuildError: std::error::Error` is not satisfied
  --> .../ethereum/src/benchmarking.rs:50:38
   |
50 |     let tx = TxEnv::builder().build()?;
   |                               -------^ the trait `std::error::Error` is not implemented for `revm::revm_context::tx::TxEnvBuildError`
   |                               |
   |                               this has type `Result<_, revm::revm_context::tx::TxEnvBuildError>`
   |
   = note: the question mark operation (`?`) implicitly performs a conversion on the error value using the `From` trait
   = note: required for `anyhow::Error` to implement `std::convert::From<revm::revm_context::tx::TxEnvBuildError>`
```

--- a/crates/context/src/tx.rs
+++ b/crates/context/src/tx.rs
@@ -612,9 +612,7 @@ impl core::fmt::Display for TxEnvBuildError {
             Self::MissingGasPriorityFeeForEip1559 => {
                 f.write_str("missing gas priority fee for EIP-1559")
             }
-            Self::MissingBlobHashesForEip4844 => {
-                f.write_str("missing blob hashes for EIP-4844")
-            }
+            Self::MissingBlobHashesForEip4844 => f.write_str("missing blob hashes for EIP-4844"),
             Self::MissingAuthorizationListForEip7702 => {
                 f.write_str("missing authorization list for EIP-7702")
             }

